### PR TITLE
Bug 1909864: add missing /usr/local/e2e-aws mount

### PR DESCRIPTION
**Pull request by miabbott**
All of the promote jobs have this mount for `/usr/local/e2e-aws`,
except for the 4.5 promote job.

Add this mount into the 4.5 job spec to make it match the other
promote jobs.

### CODE REVIEW

File: core-services/template-deprecation/_allowlist.yaml [hunk 1]
```diff
@@ -706,6 +706,9 @@ templates:
     unknown_blocker:
       description: unknown
       jobs:
+        promote-release-openshift-machine-os-content-e2e-aws-4.5:
```
Comment (alvaroaleman) [inline core-services/template-deprecation/_allowlist.yaml:709]: Why is this needed to modify an existing job?

Comment (petr-muller) [reply core-services/template-deprecation/_allowlist.yaml:709]: Because previously the job did not mount the template at all although it attempted to use it (see the added `prow-job-cluster-launch-installer-e2e` CM mount). `template-deprecator` uses template CM to identify template-using jobs so this looks legit to me

**Review by alvaroaleman — APPROVED** [review]

**Comment by miabbott** [comment]
Thank you @alvaroaleman for the quick response :pray:
